src/protocols/secure_channel/tests/TestCheckinMsg.cpp

```diff
@@ -56,17 +56,24 @@ void TestCheckin_Generate(nlTestSuite * inSuite, void * inContext)
     {
         const ccm_128_test_vector & test = *testPtr;
 
-        Symmetric128BitsKeyByteArray keyMaterial;
-        memcpy(keyMaterial, test.key, test.key_len);
+        // Two disctint key material buffers to force the PSA unit tests to create two different Key IDs
+        Symmetric128BitsKeyByteArray aesKeyMaterial;
+        memcpy(aesKeyMaterial, test.key, test.key_len);
 
-        Aes128KeyHandle keyHandle;
-        NL_TEST_ASSERT_SUCCESS(inSuite, keystore.CreateKey(keyMaterial, keyHandle));
+        Symmetric128BitsKeyByteArray hmacKeyMaterial;
+        memcpy(hmacKeyMaterial, test.key, test.key_len);
+
+        Aes128KeyHandle aes128KeyHandle;
+        NL_TEST_ASSERT_SUCCESS(inSuite, keystore.CreateKey(aesKeyMaterial, aes128KeyHandle));
+
+        Hmac128KeyHandle hmac128KeyHandle;
+        NL_TEST_ASSERT_SUCCESS(inSuite, keystore.CreateKey(hmacKeyMaterial, hmac128KeyHandle));
 
         // Validate that counter change, indeed changes the output buffer content
         counter = 0;
         for (uint8_t j = 0; j < 5; j++)
         {
-            err = CheckinMessage::GenerateCheckinMessagePayload(keyHandle, counter, userData, outputBuffer);
+            err = CheckinMessage::GenerateCheckinMessagePayload(aes128KeyHandle, hmac128KeyHandle, counter, userData, outputBuffer);
```

Comment:
Here we verify that the generate did not trigger any error, would it be possible to confirm that the actual output content has the expected value (given we control the random counter)?

Reply:
As discussed, overhaul will be in a follow up PR (unfortunately)